servers/src/mining/stratumserver.rs

```diff
@@ -574,16 +575,21 @@ impl Handler {
 					// set a new deadline for rebuilding with fresh transactions
 					deadline = Utc::now().timestamp() + config.attempt_time_per_block as i64;
 
-					self.workers.update_block_height(new_block.header.height);
-					self.workers
-						.update_network_difficulty(state.current_difficulty);
-
+					// Manage list of versions of this heights block
 					if clear_blocks {
 						state.current_block_versions.clear();
 					}
 					state.current_block_versions.push(new_block);
-					// Send this job to all connected workers
+
+					// Update the mining stats
+					self.workers
+						.update_block_height(state.current_block_versions[0].header.height);
+					let difficulty = state.current_block_versions[0].header.total_difficulty()
+						- head.total_difficulty;
+					self.workers.update_network_difficulty(difficulty.to_num());
```

Comment:
Same here those two lines can be replaced by:
```suggestion
					self.workers
 						.update_network_difficulty(state.current_difficulty);
```

Reply:
> Few comments. Also don't think it' super clean to have both the `edge_bits` and `network_hashrate` potentially oscillating for every share (i.e. mining c29 and c32 simultaneously). But we need to also consider that c29 has 4 months left.

I think it is possible to mine C33(+?) so we still need some way to show multiple pow size and rate.  Showing data for the last submitted share (oscillating) is better than showing bogus data, IMO.